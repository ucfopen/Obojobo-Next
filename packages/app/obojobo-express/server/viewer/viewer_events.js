const oboEvents = oboRequire('server/obo_events')
const viewerState = oboRequire('server/viewer/viewer_state')
const VisitModel = oboRequire('server/models/visit')
const db = oboRequire('/server/db')
const logger = oboRequire('/server/logger')

oboEvents.on('client:nav:open', event => {
	return setNavOpen(event.userId, event.draftId, event.contentId, true, event.visitId)
})

oboEvents.on('client:nav:close', event => {
	return setNavOpen(event.userId, event.draftId, event.contentId, false, event.visitId)
})

oboEvents.on('client:nav:setRedAlert', event => {
	const userId = event.userId
	const draftId = event.draftId
	const isRedAlertEnabled = event.payload.to
	return db.none(
		`
				INSERT INTO red_alert_status
				(user_id, draft_id, is_enabled)
				VALUES($[userId], $[draftId], $[isRedAlertEnabled])
				ON CONFLICT (user_id, draft_id) DO UPDATE
				SET is_enabled = $[isRedAlertEnabled]
				WHERE red_alert_status.user_id = $[userId] AND
				red_alert_status.draft_id = $[draftId]
			`,
		{
			userId,
			draftId,
			isRedAlertEnabled
		}
	)
})

const setNavOpen = (userId, draftId, contentId, value, visitId) => {
	return VisitModel.fetchById(visitId).then(visit => {
		viewerState.set(userId, draftId, contentId, 'nav:isOpen', 1, value, visit.resource_link_id)
	})
}

module.exports = (req, res, next) => {
	next()
}
