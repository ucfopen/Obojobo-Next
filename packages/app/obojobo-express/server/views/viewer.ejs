<!DOCTYPE html>
<html lang="en">

<head>
	<%
			const title = 'Obojobo Next Document Viewer'
			const css = [
				webpackAssetPath('viewer.css'),
			]
			const headerJs = []
			const footerJs = [
				assetForEnv('//unpkg.com/react@16.13.1/umd/react.$[development|production.min].js'),
				assetForEnv('//unpkg.com/react-dom@16.13.1/umd/react-dom.$[development|production.min].js'),
				assetForEnv('//unpkg.com/underscore@1.11.0/underscore$[-min].js'),
				assetForEnv('https://unpkg.com/backbone@1.3.3/backbone$[-min].js'),
				webpackAssetPath('viewer.js')
			]
			const fonts = ['//fonts.googleapis.com/css?family=Libre+Franklin:400,400i,700,700i,900,900i|Roboto+Mono:400,400i,700,700i|Noto+Serif:400,400i,700,700i']
		%>
		<%- include('./partials/head', {title, css, fonts, headerJs}) %>
</head>

<body>
	<div id="viewer-app"></div>
	<div id="viewer-app-loading">
		<%- include('./partials/loading') %>
	</div>
	<%
		if (typeof inlines !== 'undefined')
		{
			inlines.forEach(function(inline)
			{ %>
		<script><% - js %></script>
		<% })
		}

		if (typeof footerJs !== 'undefined')
		{
			footerJs.forEach(function(href) { %>
			<script src="<%= href %>"></script>
			<% })
		}
	%>
				<script>window.__oboViewerRender()</script>
</body>

</html>
