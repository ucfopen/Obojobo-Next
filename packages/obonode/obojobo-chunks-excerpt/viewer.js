import Common from 'obojobo-document-engine/src/scripts/common'
import ViewerComponent from './viewer-component'
import adapter from './adapter'

Common.Registry.registerModel('ObojoboDraft.Chunks.Excerpt', {
	adapter,
	componentClass: ViewerComponent,
	default: true,
	type: 'chunk'
})
