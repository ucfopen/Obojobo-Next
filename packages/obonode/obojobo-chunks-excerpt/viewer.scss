@import '../obojobo-document-engine/src/scss';

.obojobo-draft--chunks--excerpt {
	pre {
		font-size: 1em;
		padding-left: 2em;
		padding-right: 2em;
	}

	.text {
		display: block;
	}
}
