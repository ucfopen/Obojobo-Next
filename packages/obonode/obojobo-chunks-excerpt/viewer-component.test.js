import React from 'react'
import renderer from 'react-test-renderer'

import Code from './viewer-component'
import OboModel from 'obojobo-document-engine/src/scripts/common/models/obo-model'

const chunkJSON = {
	id: 'id',
	type: 'ObojoboDraft.Chunks.Code',
	content: {
		textGroup: [
			{
				text: {
					value: 'Example Text'
				}
			}
		]
	}
}

require('./viewer') // used to register this oboModel

describe('Code', () => {
	test('Code component', () => {
		const model = OboModel.create(chunkJSON)
		const moduleData = {
			focusState: {}
		}
		const component = renderer.create(<Code model={model} moduleData={moduleData} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})
})
