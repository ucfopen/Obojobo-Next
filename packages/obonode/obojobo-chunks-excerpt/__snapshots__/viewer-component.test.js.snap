// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code Code component 1`] = `
<div
  className="component"
  data-focussed={false}
  data-id="id"
  data-obo-component={true}
  data-type="ObojoboDraft.Chunks.Code"
  id="obo-id"
  tabIndex={-1}
>
  <div
    className="text-chunk obojobo-draft--chunks--code pad"
  >
    <pre>
      <code>
        <span
          className="text align-left"
          data-group-index={0}
          data-indent={0}
        >
          <span>
            <span>
              <span>
                Example Text
              </span>
            </span>
          </span>
        </span>
      </code>
    </pre>
  </div>
</div>
`;
